### PR TITLE
RPKI validation and multiprocessing (fixes #197, #280)

### LINKED ISSUE #197: RPKI Origin Validation in context of IRR
Allow a new type of data (RPKI ROAs) to 'drown out' IRR data sources. The goal is to have RPKI data supersede IRR data. By using RPKI data we can combat stale or fraudulent IRR proxy route registrations.

Consider the following:

```
$ whois -h mockup 192.0.2.0
route: 192.0.2.0/24
origin: AS1
source: NTTCOM

route: 192.0.2.0/25
origin: AS2
source: NTTCOM

roa: 192.0.2.0/24
maxlength: 24
origin: AS2
source: RPKI # RIPE TAL
```

What I'd like to see is that `!gAS1` (or `!aAS1`) will _not_ return `192.0.2.0/24 192.0.2.0/25` because of the existence of the RPKI ROA entry covering `192.0.2.0/24 maxlen 24`.

The `192.0.2.0/24AS1` entry is hidden because it conflicts with the `roa:` entry, or phrased differently because there is no roa that permits that origin for that prefix. The procedure to validate `route:` objects against RPKI ROAs can be taken from RFC 6811 section 2.

The RPKI ROA database is encoded in a JSON file and can be downloaded (for example) from https://rpki.gin.ntt.net/api/export.json, this table is expected to refresh every 20 minutes.

A new data type in the IRRd data model should introduces, the 'roa'.

From the downloaded RPKI ROA data an IRR-like source should be synthesised. `!g`, `!a` etc should return routes like `route:` objects would. The synthesised RPKI source should be enabled by default.

The RPKI Origin Validation process should be applied on all the entry points: NRTM, submissions to authoritative, or dump loads. If a `route:` object is encountered that conflicts with all ROAs it should be marked as 'invalid'.

All (or a cleverly decided subset) `route:` objects must be re-validated when `export.json` contains new roas or roas are deleted.

### PR DESCRIPTION
**Pull request by mxsasha**
- [x] Send notifications to owners of auth objects marked RPKI-invalid
- [x] Preload source filtering as implemented in in-memory already
- [x] Look into double preloading
- [x] Rename unknown to not-found
- [x] Extend ACL tests
- [x] Modify `!f` command to be more extendable in the future
- [x] Verify whether RPKI invalid excluded/included queries are used correctly everywhere
- [x] Verify behaviour for modifying RPKI-invalid objects
- [x] Add note to update rejection notice about ROA refresh time.
- [x] Fix possible requests missing from dependencies
- [x] Research issues regarding setcap, privilege drop and socket binding
- [x] Errors in socket binding or other server startup errors may cause dead or zombie processes
- [x] Ensure docs are clear about ROA refresh time vs early NRTM.
- [x] Reflect RPKI state correctly in NRTM for other mirrors and database exports
- [x] Move RPKI exclusion to be defined per source
- [x] Make remarks field in pseudo-IRR replacable (with string replace for e.g. $AS)
- [x] Prevent accidental email sending to live contacts
- [x] Add RPKI validation status to status interface
- [x] Review code
- [x] Review docs
- [x] Additional testing on irrd02

#270 will remain out of the scope of this PR, as it's already quite large.

### CODE REVIEW

**Comment by mxsasha** [comment]
@job who exactly would you like to have notified when objects are flagged as invalid? The e-mail attribute on the tech-c? admin-c?

Assuming this feature is limited to NTTCOM, that source currently has 46351 RPKI-invalid objects, which will be flagged and prompt notification upon enabling RPKI validation. Those have 46351 objects have 9 different admin-c's, and 13 different tech-c's. I suspect at least one from each set is not a valid object reference, so we can't resolve it to an e-mail address. We'd aggregate per recipient I assume, so upon enabling RPKI in IRRd, about a dozen emails go out, not 46000.

**Comment by job** [comment]
Yes, both `admin-c` and `tech-c` (deduplicated)

File: docs/development/storage.rst [hunk 1]
```diff
@@ -122,6 +128,23 @@ For each source, a record is kept of:
...
+updates and full mirror imports.
+
+* `pk`: a random UUID, primary key of the table.
+* `prefix`: the prefix of the ROA
+* `asn`: the valid origin AS recorded in the ROA (can be zero)
+* `max_length`: the max prefix length the ROA allows
+* `trust_anchor`: the trust anchor for the ROA (free text)
+* `ip_version`: the IP version of `prefix`.
+
+The fields `prefix`, `asn` and `max_length` must be unique together.
```
Comment (job) [inline docs/development/storage.rst:145]: probably want to add `trust_anchor` to that unique primary key tuple too.

**Comment by job** [comment]
Some comments:

1. I'd like `roa_source` to be set by default to `https://rpki.gin.ntt.net/api/export.json`
2. Can `!fno-rpki-filter` also be made to work for `!a` and `!g` ?
3. RPKI invalids route/route6 objects should not be outputted into dumps / NRTM unless the source was listed in `rpki.validation_excluded_sources`

**Comment by job** [comment]
As discussed by phone, can a `irrd_vacuum_rpki_invalids` script be made to remove invalid cruft from the DB if its been a while

**Comment by mxsasha** [comment]
> Yes, both `admin-c` and `tech-c` (deduplicated)

@job this will not work the way we thought.

In NTTCOM, there are now 46331 RPKI-invalid route(6) objects. The data of them belonging to a handful of tech-c/admin-c's is incomplete, because route(6) objects often do not have an admin-c or tech-c. Of those 46K, 42K do not have an admin-c or tech-c. So notifying the tech-c/admin-c on the object is pointless.

All invalids, as required, have an mnt-by. These in turn should all have an admin-c/tech-c. Should we go for: email all e-mail attributes, of all tech-c and admin-c contacts, of all maintainers of an invalid object? Or something else?

File: irrd/storage/tests/test_preload.py [hunk 2]
```diff
@@ -1,106 +1,104 @@
...
-        with open(pidfile, 'w') as fh:
-            fh.write('a')
 
-        # Call the reload signal with an incorrect PID and incorrect filename
-        send_reload_signal(pidfile)
-        assert 'Attempted to send reload signal to update preloader for IRRD on PID a, but process is' in caplog.text
-        send_reload_signal(str(tmpdir) + '/invalid_pidfile')
-        assert 'but file could not be opened: [Errno 2] No such file or directory' in caplog.text
-        assert not len(mock_preload_updater.mock_calls)
+        # Listen() on redis is blocking, unblock it after setting terminate
```
Comment (mxsasha) [inline irrd/storage/tests/test_preload.py:81]: Is this a concern during shutdown?

File: irrd/updates/parser.py [hunk 3]
```diff
@@ -28,6 +32,7 @@ class ChangeRequest:
     mntners_notify: List[RPSLMntner]
 
     error_messages: List[str]
+    warning_messages: List[str]
```
Comment (mxsasha) [inline irrd/updates/parser.py:35]: We don't use warnings anymore?

File: irrd/updates/parser.py [hunk 4]
```diff
@@ -241,6 +256,7 @@ def _check_references(self) -> bool:
         of references from the new object to others matter.
         """
         if self.request_type == UpdateRequestType.DELETE and self.rpsl_obj_current is not None:
+            # TODO: shouldn't this check on rpsl_obj_current?
```
Comment (mxsasha) [inline irrd/updates/parser.py:259]: TODO

File: irrd/updates/parser.py [hunk 5]
```diff
@@ -250,14 +266,49 @@ def _check_references(self) -> bool:
...
+        """
+        Check whether there are any conflicting ROAs with the new object.
+        Result is cached, as validate() may be called multiple times,
+        but the result of this check will not change. Always returns
+        True when not in RPKI-aware mode.
+        """
+        assert self.rpsl_obj_new
+        if not get_setting('rpki.roa_source') or not self.rpsl_obj_new.rpki_relevant:
+            return True
+        if self._cached_roa_validity is not None:
```
Comment (mxsasha) [inline irrd/updates/parser.py:287]: This can be moved up two lines

**Comment by mxsasha** [comment]
Some numbers with the latest changes to mirroring: when enabling RPKI, 150.000 NRTM deletes were added to the journal, to delete invalid objects from mirrors.

The initial process took 12 minutes, later ones took 3 minutes.

For NTTCOM, this resulted in only 18 emails, because PLACEHOLDER-NTTCOM is the admin-c/tech-c for a large majority of the maintainers of RPKI invalid routes. We should consider to include `mnt-nfy` on the mntner as a contact address, as that seems to be filled for almost all mntners. Currently, the notifications have rather limited value.

NTTCOM does seem to have had a cleanup of RPKI invalids, as it dropped from 46K invalid routes to 5K.

**Pull request merged**